sailor/utils/oauth_wrapper/OAuthServiceImpl.py

```diff
@@ -40,55 +43,67 @@ def __init__(self, name, scope_config=SCOPE_CONFIG):
-        :type endpoint_url: str
-        :param method: Specifies the HTTP method
-        :type method: str
-        :param parameters: Set of parameters that needs to be send along with the request.
-        :type parameters: dict
-        """
-        if self.configured_scopes and not self.resolved_scopes:
-            try:
-                self._resolve_configured_scopes()
-            except Exception as exc:
-                warnings.warn('Could not resolve the configured scopes. Trying to continue without scopes...')
-                LOG.debug(exc, exc_info=True)
+    def request(self, method, url, **req_kwargs):
+        """Make a request using this convenience wrapper.
 
-        scope = ' '.join(self.resolved_scopes) if self.resolved_scopes else None
-        session = self._get_session(scope=scope)
+        The interface is the same as the :meth:`~requests.sessions.Session.requests` method provides.
+        Adds OData param url formatting for GET requests.
+        Requests for JSON content by default and returns JSON response by default.
+
+        Will use the currently attached session with this client or create a new one. If scopes are configured with
+        this client, will try to resolve the scopes with the auth server first before making the request.
 
-        if not parameters:
-            parameters = {}
 
-        parameters['$format'] = 'json'
-        endpoint_url = furl(endpoint_url)
-        endpoint_url.args = {**endpoint_url.args, **parameters}
-        endpoint_url = endpoint_url.tostr(query_quote_plus=False)
+        Returns
+        -------
+        A dict if the response contains JSON. Otherwise returns the response content as bytes.
 
-        LOG.debug('Calling %s', endpoint_url)
-        response = session.request(method, endpoint_url)
+        Raises
+        ------
+        RequestError
+            When the reponse is retrieved but response code is less than 400.
+        """
+        if method == 'GET':
+            parameters = dict(req_kwargs.setdefault('params', {}))
+            parameters.setdefault('$format', 'json')
+            url_obj = furl(url)
+            url_obj.args = {**url_obj.args, **parameters}
+            url = url_obj.tostr(query_quote_plus=False)
+            del req_kwargs['params']
+
+        req_kwargs.setdefault('headers', {'Accept': 'application/json'})
+
+        LOG.debug('Calling %s', url)
+        response = self.get_session().request(method, url, **req_kwargs)
         if response.ok:
             if response.headers['Content-Type'] == 'application/json':
                 return response.json()
             else:
                 return response.content
         else:
             msg = f'Request failed. Response {response.status_code} ({response.reason}): {response.text}'
-            LOG.error(msg)
             raise RequestError(msg, response.status_code, response.reason, response.text)
 
+    def get_session(self):
```

Comment:
the function names `get_session` and `_get_session` don't really make it clear what the purpose of the two functions is. Not sure what good names should be, maybe `_get_session` should be something with `maybe_cached` or so? (Or the two functions could fold into one?)
Also, is `get_session` really public?

Reply:
The whole class should not be public I guess. But that's besides the point .

First: It's important that the "scope resolving" is not in the `_get_session` method because it makes the code more convoluted.

The code was before in the fetch_endpoint_data method (now request). And that's where I've moved it again now. It makes most sense.